### PR TITLE
refactor: move app vuex store to pinia

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Removes the vuex app store module in favour of a pinia store instead.

I also tried cleaning up and improving the naming and types of app related stuff. It's not all perfect yet, but it's definitely an improvement. The most notable change here is that `extensions` inside the `appInfo` object are now of type `ApplicationFileExtension[]`, since they have nothing to do with our extension library, but are file extensions.

Also, there is no such thing as a `newFileHandler`. This is basically a `ApplicationFileExtension` with an added `newFileMenu` that includes a `menuTitle`.

### Breaking changes for developers

* Removes the `announceExtensions` method inside the app's `ready` hook, which could be used to register file editors. Apparently it was not used by any app because there is already another way to register file editors, which is via the `extensions` property inside the `appInfo` object.
* Removes the global `$store` variable. The recommended way to use the vuex store is via `useStore`.

## Related Issue

- refs https://github.com/owncloud/web/issues/10210

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: packages/web-app-preview/src/App.vue [hunk 1]
```diff
@@ -153,6 +139,10 @@ export default defineComponent({
     const cachedFiles = ref<CachedFile[]>([])
     const folderLoaded = ref(false)
 
+    const mimeTypes = computed(() => {
+      return getMimeTypes(appsStore.externalAppConfig[appId]?.mimeTypes)
```
Comment (dschmidt) [inline packages/web-app-preview/src/App.vue:143]: Can't we use the applicationConfig prop for this?

File: packages/web-app-preview/src/index.ts [hunk 2]
```diff
@@ -1,41 +1,46 @@
...
-    mimeType,
-    routeName,
-    label: $gettext('Preview')
-  }))
-}
+    const appInfo = {
+      name: $gettext('Preview'),
+      id: appId,
+      icon: 'eye',
+      extensions: getMimeTypes(appsStore.externalAppConfig[appId]?.mimeTypes).map((mimeType) => ({
```
Comment (dschmidt) [inline packages/web-app-preview/src/index.ts:33]: and can't we use the applicationConfig from the ApplicationSetupOptions here?

File: packages/web-app-text-editor/src/index.ts [hunk 3]
```diff
@@ -50,17 +53,16 @@ export default defineWebApplication({
         }
       ]
 
-      const config = (window as any).__$store.getters.extensionConfigByAppId(appId)
-      extensions.push(...(config.extraExtensions || []).map((ext) => ({ extension: ext })))
+      const config = appsStore.externalAppConfig[appId] || {}
```
Comment (dschmidt) [inline packages/web-app-text-editor/src/index.ts:56]: and here

File: packages/web-pkg/src/apps/types.ts [hunk 4]
```diff
@@ -42,18 +42,33 @@ export interface ApplicationQuickAction {
...
+  color?: string
   id?: string
   name?: string
   icon?: string
+  iconFillType?: IconFillType
+  iconColor?: string
+  img?: string
   isFileEditor?: boolean
-  extensions?: any[]
+  extensions?: ApplicationFileExtension[]
```
Comment (dschmidt) [inline packages/web-pkg/src/apps/types.ts:71]: 😍

File: packages/web-pkg/src/composables/actions/files/useFileActionsCreateNewFile.ts [hunk 5]
```diff
@@ -208,27 +209,25 @@ export const useFileActionsCreateNewFile = ({
...
+        ext: appFileExtension.extension
       })
     }
     for (const mimeType of unref(mimetypesAllowedForCreation) || []) {
-      const openAction = false
       actions.push({
         name: 'create-new-file',
         icon: 'add',
-        handler: (args) => handler(args, mimeType.ext, openAction),
+        handler: (args) => handler(args, mimeType.ext, undefined),
```
Comment (dschmidt) [inline packages/web-pkg/src/composables/actions/files/useFileActionsCreateNewFile.ts:230]: Why the undefined?

File: packages/web-pkg/src/composables/appDefaults/useAppConfig.ts [hunk 6]
```diff
@@ -13,9 +12,8 @@ export interface AppConfigResult {
 }
 
 export function useAppConfig(options: AppConfigOptions): AppConfigResult {
-  const applicationMetaResult = useAppMeta(options)
   const applicationConfig = computed(
-    () => unref(applicationMetaResult.applicationMeta).config || {}
+    () => options.appsStore.externalAppConfig[options.applicationId] || {}
```
Comment (dschmidt) [inline packages/web-pkg/src/composables/appDefaults/useAppConfig.ts:16]: Is this something else or just removing an indirection?

File: packages/web-pkg/src/composables/appDefaults/useAppMeta.ts
Comment (dschmidt) [inline packages/web-pkg/src/composables/appDefaults/useAppMeta.ts:1]: 😍

File: packages/web-runtime/src/container/application/classic.ts [hunk 7]
```diff
@@ -128,7 +127,8 @@ export const convertClassicApplication = async ({
     supportedLanguages
   })
 
-  await store.dispatch('registerApp', applicationScript.appInfo)
+  const appsStore = useAppsStore()
+  appsStore.registerApp(applicationScript.appInfo)
```
Comment (dschmidt) [inline packages/web-runtime/src/container/application/classic.ts:131]: Exciting - why was this async before?

File: packages/web-runtime/src/container/application/index.ts [hunk 8]
```diff
@@ -122,7 +122,7 @@ export const buildApplication = async ({
     if (!isObject(applicationScript.appInfo) && !applicationScript.setup) {
       throw new RuntimeError('next applications not implemented yet, stay tuned')
```
Comment (dschmidt) [inline packages/web-runtime/src/container/application/index.ts:123]: We should get rid of Next Applications as a concept... Classic and Next Applications are somewhat converging anyhow

**Review by dschmidt — APPROVED** [review]
😍 😍 😍

File: packages/web-app-preview/src/App.vue (hunk 1 above)
Comment (JammingBen) [reply packages/web-app-preview/src/App.vue:143]: As discussed, we need this to be wrapped in `AppWrapper` first.

File: packages/web-pkg/src/composables/actions/files/useFileActionsCreateNewFile.ts (hunk 5 above)
Comment (JammingBen) [reply packages/web-pkg/src/composables/actions/files/useFileActionsCreateNewFile.ts:230]: As discussed, before it was `false`, so this is the same behaviour but correct typing.

File: packages/web-pkg/src/composables/appDefaults/useAppConfig.ts (hunk 6 above)
Comment (JammingBen) [reply packages/web-pkg/src/composables/appDefaults/useAppConfig.ts:16]: As discussed, config was removed from the app meta.

File: packages/web-runtime/src/container/application/classic.ts (hunk 7 above)
Comment (JammingBen) [reply packages/web-runtime/src/container/application/classic.ts:131]: As discussed, because the signature of `store.dispatch` is async.

**Pull request merged**